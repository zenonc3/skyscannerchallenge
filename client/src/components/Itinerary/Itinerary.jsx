import BpkTicket from 'bpk-component-ticket';
import React from 'react';
import Pricing from '../Pricing';
import Leg from '../Leg';
import STYLES from './Itinerary.scss';

const c = className => STYLES[className] || 'UNKNOWN';

const Itinerary = ({ outbound, inbound, pricing }) => {
  const pricingComponent = <Pricing {...pricing} />;
  return (
    <div className={c('Itinerary__container')}>
      <BpkTicket stub={pricingComponent} vertical>
        <Leg {...outbound} />
        <Leg {...inbound} />
      </BpkTicket>
    </div>
  );
};

Itinerary.propTypes = {
  outbound: Leg.propTypes.isRequired,
  inbound: Leg.propTypes.isRequired,
  pricing: Pricing.propTypes.isRequired,
};

export default Itinerary;
