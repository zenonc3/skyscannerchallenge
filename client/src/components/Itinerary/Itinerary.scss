.Itinerary {
  &__container {
    margin-bottom: 10pt;
  }
}
