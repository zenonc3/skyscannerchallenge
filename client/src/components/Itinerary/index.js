import Itinerary from './Itinerary';

export default Itinerary;
