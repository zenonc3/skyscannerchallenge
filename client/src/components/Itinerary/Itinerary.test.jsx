import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Itinerary from './Itinerary';


describe('Itinerary', () => {
  const dummyLeg = {
    origin: '',
    destination: '',
    departure: '',
    arrival: '',
    duration: 10,
    numStops: 1,
    carrier: '',
  };
  const dummyPricing = {
    price: 123,
    agent: '',
    bookingUrl: '',
  };
  const dummyItinerary = {
    outbound: dummyLeg,
    inbound: dummyLeg,
    pricing: dummyPricing,
  };
  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Itinerary
      outbound={dummyItinerary.outbound}
      inbound={dummyItinerary.inbound}
      pricing={dummyItinerary.pricing}
    />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

