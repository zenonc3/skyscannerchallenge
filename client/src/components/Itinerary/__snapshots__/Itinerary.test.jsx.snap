// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itinerary should render correctly 1`] = `
<div
  className="Itinerary__container"
>
  <BpkTicket
    className={null}
    href={null}
    padded={true}
    stub={
      <Pricing
        agent=""
        bookingUrl=""
        price={123}
      />
    }
    stubClassName={null}
    stubProps={Object {}}
    vertical={true}
    withNotches={true}
  >
    <Leg
      arrival=""
      carrier=""
      departure=""
      destination=""
      duration={10}
      numStops={1}
      origin=""
    />
    <Leg
      arrival=""
      carrier=""
      departure=""
      destination=""
      duration={10}
      numStops={1}
      origin=""
    />
  </BpkTicket>
</div>
`;
